### PR TITLE
fix: response error interceptor broken

### PR DESCRIPTION
**Pull request by luddd3**
## This relates to...

The response-error interceptor

## Rationale

The interceptor is currently broken since Request no longer accepts `throwOnError` as an option.

## Changes

* Expose the interceptor in the same manner as the others.
* Remove `throwOnError` as it is considered an invalid argument.
* Removed broken/incomplete tests

## Status

- [x] I have read and agreed to the [Developer's Certificate of Origin][cert]
- [x] Tested
- [ ] Benchmarked (**optional**)
- [x] Documented
- [x] Review ready
- [ ] In review
- [ ] Merge ready

[cert]: https://github.com/nodejs/undici/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by metcoder95 — APPROVED** [review]

File: lib/interceptor/response-error.js [hunk 1]
```diff
@@ -84,6 +84,10 @@ class Handler extends DecoratorHandler {
...
 
-module.exports = (dispatch) => (opts, handler) => opts.throwOnError
-  ? dispatch(opts, new Handler(opts, { handler }))
-  : dispatch(opts, handler)
+module.exports = () => {
+  return (dispatch) => {
+    return function Intercept (opts, handler) {
+      return dispatch(opts, new ResponseErrorHandler(opts, { handler }))
+    }
+  }
```
Comment (Uzlopak) [inline lib/interceptor/response-error.js:92]: Hmm, this makes me having stomach ache

**Comment by Uzlopak** [comment]
@metcoder95 

When we are already touching this file massively, should we extract the handler into a new file and put it into the handler folder?

File: lib/interceptor/response-error.js (hunk 1 above)
Comment (luddd3) [reply lib/interceptor/response-error.js:92]: Care to elaborate or give a suggestion?

**Comment by metcoder95** [comment]
It depends on what we do with that handler.
If the handler remains used within the interceptor only, let's keep it close to it; otherwise, let's add it into a separate handler file for easier management.

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**